### PR TITLE
fix: Allow Relative Paths in Page Builder Links.

### LINKED ISSUE #1260: Page Builder link validation is too restrictive.
## This is:

- Feature request

## Detailed Description
When specifying the URL for a link (in a button or a menu item, for example), Page Builder requires a full URL. 

## Possible Solution
Ideally, it should allow anything that the href property of the a tag supports (#, relative, absolute etc). I imagine the value should also be sanitized server(less)-side before being stored, though.

**Note:** Page builder [Docs](http://docs.webiny.com/docs/webiny-apps/page-builder/getting-started)

### PR DESCRIPTION
**Pull request by sccalabr**
## Related Issue

Closes #1260

## Your solution

The solution is to pass in allowRelative flag in the validator.
## How Has This Been Tested?

Existing tests in the url.test.js test the allowRelative flag.

### CODE REVIEW

**Comment by Ashu96** [comment]
Hey @sccalabr 
You can check the source files using `yarn run prettier:check` command.
And then run `yarn run prettier <file path>` to fix it.

**Comment by sccalabr** [comment]
@Ashu96 I get this when I try to run the command 

$ prettier "packages/**/src/**/*.{js,ts,tsx}" --check --config .prettierrc.js
'prettier' is not recognized as an internal or external command,  
operable program or batch file.
error Command failed with exit code 1.

How do I set up prettier?

**Comment by Ashu96** [comment]
Hey @sccalabr, sorry it didn't work.

That's how we use it. One thing to check here.
Please make sure to follow the [local setup](https://github.com/webiny/webiny-js/blob/master/docs/CONTRIBUTING.md#local-setup) for the project.

**Comment by sccalabr** [comment]
@Ashu96 is this project usually set up on windows or linux?

**Comment by Ashu96** [comment]
It's actively used and developed in Linux, Windows and MacOS.

**Comment by sccalabr** [comment]
Tests are passing locally

yarn run v1.22.5
$ jest --config jest.config.js packages/validation
 PASS   validation  packages/validation/__tests__/eq.test.js (7.167s)
 PASS   validation  packages/validation/__tests__/creditCard.test.js (7.196s)
 PASS   validation  packages/validation/__tests__/lte.test.js (7.2s)
 PASS   validation  packages/validation/__tests__/maxLength.test.js (7.31s)
 PASS   validation  packages/validation/__tests__/errors.test.js (7.328s)
 PASS   validation  packages/validation/__tests__/minLength.test.js (7.38s)
 PASS   validation  packages/validation/__tests__/multipleValidators.test.js (7.338s)
 PASS   validation  packages/validation/__tests__/invalidValidators.test.js
 PASS   validation  packages/validation/__tests__/gte.test.js
 PASS   validation  packages/validation/__tests__/json.test.js
 PASS   validation  packages/validation/__tests__/phone.test.js
 PASS   validation  packages/validation/__tests__/password.test.js
 PASS   validation  packages/validation/__tests__/validatorRegistration.test.js
 PASS   validation  packages/validation/__tests__/required.test.js
 PASS   validation  packages/validation/__tests__/integer.test.js
 PASS   validation  packages/validation/__tests__/numeric.test.js
 PASS   validation  packages/validation/__tests__/in.test.js
 PASS   validation  packages/validation/__tests__/number.test.js
 PASS   validation  packages/validation/__tests__/gt.test.js
 PASS   validation  packages/validation/__tests__/validationError.test.js
 PASS   validation  packages/validation/__tests__/create.test.js
 PASS   validation  packages/validation/__tests__/lt.test.js
 PASS   validation  packages/validation/__tests__/empty.test.js
 PASS   validation  packages/validation/__tests__/syncAsyncValidation.test.js
 PASS   validation  packages/validation/__tests__/url.test.js

Test Suites: 26 passed, 26 total
Tests:       118 passed, 118 total
Snapshots:   0 total
Time:        14.911s, estimated 24s

File: packages/validation/__tests__/url.test.js [hunk 1]
```diff
@@ -55,4 +55,13 @@ describe("url test", () => {
             true
         );
     });
+
+    it("should pass - href URL", () => {
```
Comment (Ashu96) [inline packages/validation/__tests__/url.test.js:59]: I'm interested to know how the `allowHref` behave when a string containing white space is passed. For example,

- `mailto:abc @ gmail.com`
- `#hello   world`

IMHO we should at least also handle that URL is a continuous string without any whitespace.
What do you think? @sccalabr

**Review by Ashu96 — CHANGES_REQUESTED** [review]
Good work @sccalabr :+1: 

I've left one comment about the test inputs, please check it out :slightly_smiling_face:

File: packages/validation/__tests__/url.test.js (hunk 1 above)
Comment (sccalabr) [reply packages/validation/__tests__/url.test.js:59]: sure we can do that.

**Comment by sccalabr** [comment]
image

**Review by Ashu96 — APPROVED** [review]
Nicely done @sccalabr :+1: 

We'll merge it soon :slightly_smiling_face:

**Pull request merged**